compiler/pash_runtime.py

```diff
@@ -636,9 +636,14 @@ def add_eager(eager_input_id, graph, fileIdGen, intermediateFileIdGen, use_dgsh_
         ## TODO: seperate to better use dgsh-tee params and maybe deprecate eager
         eager_node = dgsh_tee.make_dgsh_tee_node(eager_input_id, new_id)
     else:
-        ## TODO: Remove the line below if eager creates its intermediate file
+        ## TODO: Remove these lines below if eager creates its intermediate file
         ##       on its own.
-        intermediate_fid = intermediateFileIdGen.next_ephemeral_file_id()
+        intermediate_fid = intermediateFileIdGen.next_file_id()
```

Comment:
Thanks a lot for this change! It might be better if these lines (642 - 646) could be encapsulated behind a `FileId` method (in `definitions/ir/file_id.py`). One clean way I can see of doing it is by having an optional `fifo=True` argument in `next_ephemeral_file_id`. And then we could modify [this function](https://github.com/binpash/pash/blob/main/compiler/definitions/ir/file_id.py#L48) to be named `get_ephemeral_suffix`, which could check whether `fifo` is True or False, and create a different suffix accordingly. Another way, could be to create a different `TemporaryFileResource` and just create that here (and then add all the necessary serialization/to_ast code for it in `file_id.py`.